tests/src/ut_test_kvp.c

```diff
@@ -130,6 +140,159 @@ void test_ut_kvp_open( void )
     ut_kvp_close(pInstance);
 }
 
+int read_file(const char* filename, test_ut_memory_t* pInstance)
+{
+    FILE *file = fopen(filename, "r+");
+    if (file == NULL)
+    {
+        UT_LOG_ERROR("fopen");
+        pInstance->length = 0;
+        return -1;
+    }
+
+    // Negative Offset: this check should fail for zero length file
+    if (fseek(file, -1, SEEK_END) != 0)
+    {
+        UT_LOG_ERROR("zero length file\n");
+        fclose(file);
+        pInstance->length = 0;
+        return -1;
+    }
+
+    // Seek to the end to determine the file size
+    if (fseek(file, 0, SEEK_END) != 0)
+    {
+        UT_LOG_ERROR("fseek");
+        fclose(file);
+        pInstance->length = 0;
+        return -1;
+    }
+
+    pInstance->length = ftell(file);
+    if (pInstance->length < 0)
+    {
+        UT_LOG_ERROR("ftell");
+        fclose(file);
+        pInstance->length = 0;
+        return -1;
+    }
+    rewind(file);
+
+    // Allocate memory for the file contents
+    pInstance->buffer = malloc(pInstance->length + 1);
+    if (!pInstance->buffer)
+    {
+        UT_LOG_ERROR("malloc");
+        fclose(file);
+        pInstance->length = 0;
+        return -1;
+    }
+
+    // Read the file into the buffer
+    if (fread(pInstance->buffer, 1, pInstance->length, file) != pInstance->length)
+    {
+        UT_LOG_ERROR("fread");
+        free(pInstance->buffer);
+        fclose(file);
+        pInstance->length = 0;
+        return -1;
+    }
+
+    // Null-terminate the buffer
+    pInstance->buffer[pInstance->length] = '\0';
+
+    fclose(file);
+    return 0;
+}
+
+void test_ut_kvp_open_memory( void )
+{
+    ut_kvp_instance_t *pInstance = NULL;
+    ut_kvp_status_t status;
+    test_ut_memory_t pMemoryInstance;
+
+    /* Negative Test */
+    UT_LOG_STEP("ut_kvp_openMemory( NULL, NULL, -1 )");
+    status = ut_kvp_openMemory( NULL, NULL, -1);
+    UT_ASSERT( status == UT_KVP_STATUS_INVALID_INSTANCE );
+
+    /* Positive Test */
```

Comment:
```c
    /* Positive Test */
    UT_LOG_STEP("ut_kvp_createInstance");
    pInstance = ut_kvp_createInstance();
    UT_ASSERT( pInstance != NULL );

    /* Negative Test*/
    UT_LOG_STEP("ut_kvp_close( pInstance ) - Not been opened");
    ut_kvp_close(pInstance);
```

not sure I see the point of this, your function should only be testing openMemory that's what the purpose is...

The above test should have been done elsewhere already when you get to this point.